### PR TITLE
DSD-332: iconography style guide

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes [DSD-332](https://jira.nypl.org/browse/DSD-332)

## **This PR does the following:**
- Adds Style Guide `Iconography`

### Front End Review:
- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/Iconography/Iconography.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      type: "figma",
+      url:
+        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=10975%3A16",
+    },
+  }}
+/>
+
+# Iconography
+
+Use the NYPL Design System [Icon](https://nypl.github.io/nypl-design-system/storybook-static/?path=/docs/icon--control-props) component to render and control icons and select NYPL logos as `<svg>` elements within consuming apps. Below are a few example icons.
```
Comment (EdwinGuzman) [inline src/components/Iconography/Iconography.stories.mdx]: Can you add that you can select an icon name by using the `IconNames` or `LogoNames` enums? I would have imagined that the `name="check"` prop would throw an error...

File: src/components/Iconography/Iconography.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      <Icon decorative size={IconSizes.xxxlarge} name="logo_brooklyn" />
+    </span>
+  </Story>
+</Preview>
+
+The DS utilizes the `Filled` icons available in the [Material UI Design System](https://material-ui.com/components/material-icons/).
+
+### Size
+
+Icons in NYPL projects can be rendered at the following sizes.
```
Comment (EdwinGuzman) [inline src/components/Iconography/Iconography.stories.mdx]: Can you write that the sizes are available from the `IconSizes` enum? It's in the preview when you click on "code" but I think being explicit here is good.

File: src/components/Iconography/Iconography.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+| `small`    | `0.875rem` (14px) | n/a           |
+| `medium`   | `1.125rem` (18px) | n/a           |
+| `large`    | `1.5rem` (24px)   | `--space-m`   |
+| `xlarge`   | `2rem` (32px)     | `--space-l`   |
+| `xxlarge`  | `3rem` (48px)     | `--space-xl`  |
+| `xxxlarge` | `4rem` (64px)     | `--space-xxl` |
+
+### Color
+
+Icons in NYPL projects should be rendered with one color. Color options are built into the Icon component.
```
Comment (EdwinGuzman) [inline src/components/Iconography/Iconography.stories.mdx]: Can you include that the colors are available through the `IconColors` enum?

**Review by EdwinGuzman — COMMENTED** [review]
Looks great just minor comments.

File: src/components/Iconography/Iconography.stories.mdx (hunk 1 above)
Comment (bigfishdesign13) [reply src/components/Iconography/Iconography.stories.mdx]: I added additional verbiage about the enum values.

File: src/components/Iconography/Iconography.stories.mdx (hunk 2 above)
Comment (bigfishdesign13) [reply src/components/Iconography/Iconography.stories.mdx]: I added additional verbiage about the enum values.

File: src/components/Iconography/Iconography.stories.mdx (hunk 3 above)
Comment (bigfishdesign13) [reply src/components/Iconography/Iconography.stories.mdx]: I added additional verbiage about the enum values.

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**